pkg/api/login_oauth.go

```diff
@@ -306,42 +306,42 @@ func hashStatecode(code, seed string) string {
 	return hex.EncodeToString(hashBytes[:])
 }
 
-type loginError struct {
-	httpStatus    int
-	publicMessage string
-	err           error
+type LoginError struct {
+	HttpStatus    int
+	PublicMessage string
+	Err           error
 }
 
-type loginInformation struct {
-	action      string
-	user        *models.User
-	extUserInfo *models.ExternalUserInfo
+type LoginInformation struct {
```

Comment:
```suggestion
type loginInformation struct {
```

Only the inner fields have to be public 🙂

Reply:
I need to access it from the `saml` package as I do pretty much the same there.